app/models/outbound_webhook.rb

```diff
@@ -63,19 +63,24 @@ def as_json(*)
 
   private
 
+  def poll_period
+    @poll_period ||= Integer(ENV['OUTBOUND_WEBHOOK_POLL_PERIOD'] || '30')
+  end
+
   def post_hook(deploy)
     connection.post url do |req|
       req.headers['Content-Type'] = 'application/json'
       req.body = self.class.deploy_as_json(deploy).to_json
     end
   end
 
-  # loop forever until user cancels the deploy or deploy times out
+  # loop forever, with sleeping in between, until user cancels the deploy or deploy times out
   def poll_status_url(url)
     loop do
       response = connection.get(url)
       yield response.body
-      break if response.success?
+      break if response.success? && response.status != 202
```

Comment:
```
if response.success?
  break if response.status != 202
else
  raise Samson::Hooks::UserError, "error polling status endpoint"
end
```

Reply:
Do you want to raise here? This is different from the current behaviour where it would retry for non-exception errors like 5xx.

Reply:
I'd say just raise for now ... we can deal with the edge-cases later if we have to, for now I'd assume the service we are talking to is reliable :)